src/shared/syscheck_op.c

```diff
@@ -906,6 +926,207 @@ const char *get_group(__attribute__((unused)) int gid) {
+                                  DACL_SECURITY_INFORMATION,    // Requeste DACL security information
+                                  pSecurityDescriptor,          // Pointer that receives the DACL information
+                                  &lpcbSecurityDescriptor);     // Pointer that specifies the size, in bytes
+
+    if (dwRtnCode != ERROR_SUCCESS) {
+        os_free(pSecurityDescriptor);
+        return dwRtnCode;
+    }
+
+    // Retrieve a pointer to the DACL in the security descriptor.
+    bRtnBool = GetSecurityDescriptorDacl(
+                                         pSecurityDescriptor,   // Structure that contains the DACL
+                                         &fDaclPresent,         // Indicates the presence of a DACL
+                                         &pDacl,                // Pointer to ACL
+                                         &fDaclDefaulted);      // Flag set to the value of the SE_DACL_DEFAULTED flag
+
+    if (bRtnBool == FALSE) {
+        dwErrorCode = GetLastError();
+        mwarn("GetSecurityDescriptorDacl failed. GetLastError returned: %ld", dwErrorCode);
+        os_free(pSecurityDescriptor);
+        return dwErrorCode;
+    }
+
+    // Check whether no DACL or a NULL DACL was retrieved from the security descriptor buffer.
+    if (fDaclPresent == FALSE || pDacl == NULL) {
+        mwarn("No DACL was found (all access is denied), or a NULL DACL (unrestricted access) was found.");
+        os_free(pSecurityDescriptor);
+        return -1;
+    }
+
+    // Retrieve the ACL_SIZE_INFORMATION structure to find the number of ACEs in the DACL.
+    bRtnBool = GetAclInformation(
+                                 pDacl,                 // Pointer to an ACL
+                                 &aclsizeinfo,          // Pointer to a buffer to receive the requested information
+                                 sizeof(aclsizeinfo),   // The size, in bytes, of the buffer
+                                 AclSizeInformation);   // Fill the buffer with an ACL_SIZE_INFORMATION structure
+
+    if (bRtnBool == FALSE) {
+        dwErrorCode = GetLastError();
+        mwarn("GetAclInformation failed. GetLastError returned: %ld", dwErrorCode);
+        os_free(pSecurityDescriptor);
+        return dwErrorCode;
+    }
+
+    // Loop through the ACEs to get the information.
+    for (cAce = 0; cAce < aclsizeinfo.AceCount; cAce++) {
+        // Get ACE info
+        if (GetAce(pDacl, cAce, (LPVOID*)&pAce) == FALSE) {
+            mwarn("GetAce failed. GetLastError returned: %ld", GetLastError());
+            continue;
+        }
+
+        written = copy_ace_info(pAce, permissions, perm_size);
+
+        if (written > 0) {
+            permissions += written;
+            perm_size -= written;
+
+            if (perm_size > 0) {
+                continue;
+            }
+        }
+    }
+
+    return ERROR_SUCCESS;
+}
+
+unsigned int FILETIME_to_POSIX(FILETIME ft) {
```

Comment:
This function is duplicated in the library function `get_windows_file_time_epoch(FILETIME ft)` at time_op.c

Reply:
Done 7968a28